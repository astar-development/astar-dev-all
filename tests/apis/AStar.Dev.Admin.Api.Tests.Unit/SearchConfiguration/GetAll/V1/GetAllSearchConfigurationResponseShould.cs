using AStar.Dev.Utilities;
using JetBrains.Annotations;

namespace AStar.Dev.Admin.Api.SearchConfiguration.GetAll.V1;

[TestSubject(typeof(GetAllSearchConfigurationsResponse))]
public sealed class GetAllSearchConfigurationResponseShould
{
    [Fact]
    public void ContainTheExpectedValues()
    {
        var searchConfiguration = new Infrastructure.AdminDb.Models.SearchConfiguration
                                  {
                                      SiteConfigurationSlug           = "SiteConfigurationSlug",
                                      UpdatedBy                       = "UpdatedBy",
                                      UpdatedOn                       = new(2001, 1, 1),
                                      Id                              = 1,
                                      Subscriptions                   = "Subscriptions",
                                      SearchString                    = "SearchString",
                                      SearchStringSuffix              = "SearchStringSuffix",
                                      SearchStringPrefix              = "SearchStringPrefix",
                                      SearchConfigurationId           = Guid.NewGuid(),
                                      TopWallpapers                   = "TopWallpapers",
                                      SubscriptionsTotalPages         = 1,
                                      TotalPages                      = 2,
                                      StartingPageNumber              = 3,
                                      SubscriptionsStartingPageNumber = 4,
                                      TopWallpapersTotalPages         = 5,
                                      TopWallpapersStartingPageNumber = 5,
                                      UseHeadless                     = true,
                                      ImagePauseInSeconds             = 5,
                                      SlowMotionDelayInMilliseconds   = 500,
                                  };

        var sut = new GetAllSearchConfigurationsResponse(searchConfiguration);

        sut.ToJson().ShouldMatchApproved();
    }
}
