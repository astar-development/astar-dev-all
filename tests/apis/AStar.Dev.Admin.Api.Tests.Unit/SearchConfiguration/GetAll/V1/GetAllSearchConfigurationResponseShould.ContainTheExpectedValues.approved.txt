{
  "searchString": "SearchString",
  "topWallpapers": "TopWallpapers",
  "searchStringPrefix": "SearchStringPrefix",
  "searchStringSuffix": "SearchStringSuffix",
  "subscriptions": "Subscriptions",
  "imagePauseInSeconds": 5,
  "startingPageNumber": 3,
  "totalPages": 2,
  "useHeadless": true,
  "subscriptionsStartingPageNumber": 4,
  "subscriptionsTotalPages": 1,
  "topWallpapersTotalPages": 5,
  "topWallpapersStartingPageNumber": 5,
  "slowMotionDelayInMilliseconds": 500,
  "siteConfigurationSlug": "SiteConfigurationSlug",
  "updatedBy": "UpdatedBy",
  "updatedOn": "2001-01-01T00:00:00"
}